### ISSUE TITLE
Waypoints

### ISSUE THREAD

**Issue by npruehs** [npruehs/ue4-rts#120]
Allow players to queue orders for actors.

**Comment by JernejHabjan** [comment]
How can this be combined with Custom orders #122?
Example: 
User can first queue actor move, then actor attack, then actor attack - move order.

First one requires only map location, second one action + actor, third one action + map location

How is a "waypoint" defined then? combination of "action" + "map location/actor"?

How do we check if actor should execute next action (on tick?)

**Comment by npruehs** [comment]
Yes, exactly. You'd just queue whole order objects, along with all data in it. Then, you'd use an OrderFinished (and OrderCanceled) callback to check whether the queue is empty or not, dequeueing the next order if there is any.
